src/engines/kotor/walkmeshloader.cpp

```diff
@@ -0,0 +1,98 @@
+/* xoreos - A reimplementation of BioWare's Aurora engine
+ *
+ * xoreos is the legal property of its developers, whose names
+ * can be found in the AUTHORS file distributed with this source
+ * distribution.
+ *
+ * xoreos is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 3
+ * of the License, or (at your option) any later version.
+ *
+ * xoreos is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with xoreos. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/** @file
+ *  KotOR walkmesh (.wok) file loader. Ported from Supermanu's pathfinding
+ *  branch.
+ */
+
+#include "src/engines/kotor/walkmeshloader.h"
+
+namespace Engines {
+
+namespace KotOR {
+
+void WalkmeshLoader::loadFromStream(Common::SeekableReadStream &stream) {
+	stream.seek(0);
+
+	// Header format:
+	//
+	// uint8[8] - WOK version
+	// uint32 - walkmesh type
+	// uint8[48] - reserved
+	// float[3] - position
```

Comment:
You should check these, then, and throw if they're nonsensical. You should also probably catch throws (which can happen due to reading errors) where you're calling this from, too, and provide a `warning()` message.

If that WOK version is something like "WOK V1.0", you can use readUint32BE() and compare it with MKTAG('W', 'O', 'K', ' ').

Alternatively, the Aurora::AuroraFile base class provides functionality for that. My advice would have been putting this all in the KotOR walkmesh class, have it inherit from Walkmesh and Aurora::AuroraFile.

If you want to keep this as a separate class for some reason, there's no reason to create all this state. You immediately throw it away anyway. Like with the elevation evaluator, just have a `static WalkmeshData loadFromStream(Common::SeekableReadStream &stream)` method.

As is, your whole class design clashes with the current codebase. Please adapt your code to be more in line with how xoreos currently works, thanks.